riscv/clint.cc

```diff
@@ -83,6 +83,7 @@ void clint_t::increment(reg_t inc)
   }
   for (size_t i = 0; i < procs.size(); i++) {
     procs[i]->state.mip->backdoor_write_with_mask(MIP_MTIP, 0);
+    procs[i]->state.time->sync(mtime);
```

Comment:
This smells funny to me...what about initializing `time_counter_csr_t` with a `mtime_t*` that points to the clint's `mtime` variable? No sync needed then.

In any case, this call to `sync()` shouldn't be right in the middle of the two writes to MTIP; put it before or after.

Reply:
It seems more difficult and will take more code modification to pass mtime_t* from clint to the initializing time_counter_csr_t. Initializing time_counter_csr_t is done when construction of processor. Clint is initialized later in current implementation and may not exist. So I prefer current way to sync mtime.

Reply:
OK, that's reasonable. Please move the call to `sync()` either above or below the two updates to MTIP (which it seems could be easily improved to be a single update, but that's a different problem.)